jgrapht-core/src/main/java/org/jgrapht/alg/drawing/model/Rectangle.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * (C) Copyright 2018-2019, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * See the CONTRIBUTORS.md file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the
+ * GNU Lesser General Public License v2.1 or later
+ * which is available at
+ * http://www.gnu.org/licenses/old-licenses/lgpl-2.1-standalone.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR LGPL-2.1-or-later
+ */
+package org.jgrapht.alg.drawing.model;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * A base implementation for a rectangle in Euclidean space.
```

Comment:
Since rectangles are usually 2d, would it make sense to use the name Box instead?

Reply:
Yep, renamed to Box.